packages/assets-controllers/src/TokenRatesController.ts

```diff
@@ -360,24 +345,26 @@ export class TokenRatesController extends StaticIntervalPollingController<TokenR
   }
 
   #subscribeToNetworkStateChange() {
+    // TODO HERE: ------------------
     this.messagingSystem.subscribe(
       'NetworkController:stateChange',
       // TODO: Either fix this lint violation or explain why it's necessary to ignore.
       // eslint-disable-next-line @typescript-eslint/no-misused-promises
-      async ({ selectedNetworkClientId }, patches) => {
-        const {
-          configuration: { chainId, ticker },
-        } = this.messagingSystem.call(
-          'NetworkController:getNetworkClientById',
-          selectedNetworkClientId,
+      async ({ networkConfigurationsByChainId }, patches) => {
```

Comment:
(long term) do you think we could remove this `:stateChange` event with the `:networkRemoved` event?

https://github.com/MetaMask/core/blob/da63863c2c47d85362d18ba05b2465ae2109cde5/packages/network-controller/src/NetworkController.ts#L434-L441

Reply:
Out of scope, we can do this separately :)